BSP/Src/UART.c

```diff
@@ -0,0 +1,128 @@
+#include "UART.h"
+
+
+#define QUEUE_LENGTH 128
+#define QUEUE_ITEM_SIZE sizeof(uint8_t)
+
+static UART_HandleTypeDef huart4_ = {.Instance = UART4};
+UART_HandleTypeDef* huart4 = &huart4_;
+
+// Single TX queue
+static StaticQueue_t tx_queue_buffer;
+static uint8_t tx_queue_storage[QUEUE_LENGTH * QUEUE_ITEM_SIZE];
+static QueueHandle_t tx_queue;
+
+// RX queue (provided by user)
+static QueueHandle_t* rx_queue;
+
+
+static bool initialized = false;
+
+uart_status_t uart_init(UART_HandleTypeDef* handle, QueueHandle_t* rxQueue) {
+    if (handle->Instance != UART4) {
+        return UART_ERR;
+    }
+
+    // Create TX queue
+    tx_queue = xQueueCreateStatic(QUEUE_LENGTH, 
+                                 QUEUE_ITEM_SIZE,
+                                 tx_queue_storage, 
+                                 &tx_queue_buffer);
+    
+    rx_queue = rxQueue;
+
```

Comment:
Perhaps we can just allocate the rx queues locally. See Nathaniel's CAN driver for reference, but he has the little macro that allows the user to define what addresses and size for each address that they require, and then it creates the rx queues in the preprocessor step. Not sure about it though, please provide your thoughts or questions.